### PR TITLE
start of a step sequencer

### PR DESCRIPTION
**Pull request by tyleretters**

### CODE REVIEW

File: lib/track.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  self.division = n
+end
+
+function Track:set_pattern(pattern)
+  self.length = #pattern
+  self.pattern = {}
+  self.pattern_string = ""
+  for i = 1, #pattern do
+    self.pattern[i] = pattern[i]
+    self.pattern_string = self.pattern_string .. (pattern[i] and "x" or "-")
```
Comment (ryanlaws) [inline lib/track.lua:49]: Will we want a way to do e.g. dynamics, pitch, parameter changes, bursts later on? Or maybe these can be in separate sequencers that would get attached to these?

Comment (Quixotic7) [reply lib/track.lua:49]: Would be nice to have per step parameter locking.

Comment (ryanlaws) [reply lib/track.lua:49]: I think so too - maybe we could come up with a nice, simple step container to hold these properties.

File: lib/functions.lua [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  selected_track = util.clamp(selected_track + d, 1, #tracks)
+  fn.get_selected_track():select()
+  fn.get_selected_track():set_selected_index(1)
+end
+
+function fn.select_track_index(d)
+  fn.get_selected_track():set_selected_index(fn.get_selected_track():get_selected_index() + d)
+end
+
+function fn.make_random_pattern(n)
```
Comment (ryanlaws) [inline lib/functions.lua:18]: Might be interesting to use euclidean rhythms here.

**Comment by Quixotic7** [comment]
Looks like a good start to me.

**Pull request merged**